app/src/main/java/co/kr/sopt_seminar_30th/presentation/ui/adapter/HomeViewPagerAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package co.kr.sopt_seminar_30th.presentation.ui.adapter
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import co.kr.sopt_seminar_30th.presentation.ui.home.home.HomeFollowerFragment
+import co.kr.sopt_seminar_30th.presentation.ui.home.home.HomeFollowingFragment
+
+class HomeViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
+    FragmentStateAdapter(fragmentManager, lifecycle) {
+    override fun getItemCount(): Int = 2
+
+    override fun createFragment(position: Int): Fragment {
+        return when (position) {
+            FragmentPosition.FOLLOWING_FRAGMENT.position -> HomeFollowingFragment()
+            FragmentPosition.FOLLOWER_FRAGMENT.position -> HomeFollowerFragment()
+            else -> throw IndexOutOfBoundsException()
+        }
```

Comment:
position 값에 따라 보여줄 Fragment를 설정할 수 있는거군요..!
여기서 예외를 처리하는 부분도 필수적으로 작성해야하는건가요? 옳지 않은 index 값이 올 경우가 있나요?

Reply:
올바르지 않은 index 값이 올 경우는 없을 것 같다고 생각합니다. 하지만 이렇게 작성한 이유가 있습니다. position이 Following인지, Follower인지 코드적으로 명시해주고 싶었는데 when문에서 else를 강제했기 때문에 그렇습니다. 사실 위 코드는 이렇게 수정 가능합니다.

```suggestion
        return when (position) {
            FragmentPosition.FOLLOWING_FRAGMENT.position -> HomeFollowingFragment()
            else -> HomeFollowerFragment()
        }
```

하지만 이렇게 수정하게 되면 `HomeFollowerFragment`를 보여주는 position 조건이 그저 else로만 표시되어 한 눈에 조건을 알아보기 힘들다고 판단했습니다. 그래서 2개의 Fragment 각각의 position 조건을 명시하면서 when문을 사용하기 위해 저 2가지 position이 아닌 경우는 올바른 상황이 아니라고 판단하여 IndexOutOfBoundsException을 발생시키도록 했습니다.